research/runner.py

```
@@ -0,0 +1,394 @@
+      
+      if isinstance(action, (ScheduleTask, ScheduleTaskOnCores)):
+        expected_span = action.expected_span
+      
+      if expected_span is not None:
+        finish_times.append(expected_span.finish)
+    
+    return max(finish_times) if finish_times else None
+  
+  def _process_actions(self) -> None:
+    for i in range(len(self.resources)): 
+      self._process_resource_queue(i);
```

Ruff (0.15.12)

[error] Statement ends with an unnecessary semicolon

Remove unnecessary semicolon

(E703)